src/main/webapp/app/overview/course-registration/course-registration.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, OnInit } from '@angular/core';
+import { AccountService } from 'app/core/auth/account.service';
+import { Course } from 'app/entities/course.model';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { ProfileService } from 'app/shared/layouts/profiles/profile.service';
+import { matchesRegexFully } from 'app/utils/regex.util';
+import { HttpResponse } from '@angular/common/http';
+
+@Component({
+    selector: 'jhi-course-registration-selector',
+    templateUrl: './course-registration.component.html',
+})
+export class CourseRegistrationComponent implements OnInit {
+    courses: Course[];
+    public coursesToSelect: Course[] = [];
+    public userIsAllowedToRegister = false;
+    addedSuccessful = false;
+    loading = false;
+
+    constructor(private accountService: AccountService, private courseService: CourseManagementService, private profileService: ProfileService) {}
+
+    ngOnInit(): void {
+        this.loading = true;
+        this.courseService.findAllForDashboard().subscribe((res: HttpResponse<Course[]>) => {
+            this.courses = res.body!;
+            this.loadAndFilterCourses();
+        });
+        this.accountService.identity().then((user) => {
+            this.profileService.getProfileInfo().subscribe((profileInfo) => {
+                if (profileInfo) {
+                    this.userIsAllowedToRegister = matchesRegexFully(user!.login, profileInfo.allowedCourseRegistrationUsernamePattern);
```

Comment:
If you'd consider my other commend regarding moving parts to the server, we could also place this in the server.
If the user is not allowed, return no courses. In the template, you don't have special translations anyway.